### PR TITLE
Show previous month invoice amount and hours while sending the invoice

### PR DESCRIPTION
**Pull request by Gauravchamoli1997**
Targets #2676
- Firstly Inside the Invoices/Resource/views/modals folder I  create automatic-invoice-details.blade.php file and Fill the dummy data inside the modal.
- Then  Now on the invoice page for every ready-to-send invoices (project level) I create an arrow button that will open the modal when pressed and also have to include the modal page in the page.
- Now inside the modal I print the name of the project for which we have opened it.
- In the modal, set header as <Project Name> Invoice Details.
- After that I create a table inside the modal body and works as requirement.

- [x] I have performed a self-review of my own code.

### CODE REVIEW

**Comment by GauravGusain98** [comment]
Testing Feedbacks
- The table data is interchanged. You're showing previous month data in current month column and vice versa. Please correct it
- The previous invoice amount and hours are not coming correct

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comment

File: Modules/Client/Entities/Client.php [hunk 1]
```diff
@@ -107,6 +108,14 @@ public function billingDetails()
         return $this->hasOne(ClientBillingDetail::class)->withDefault();
     }
 
+    public function previousBillingDate()
+    {
+        $lastInvoiceDate = DB::table('invoices')->where('client_id', $this->id)->orderBy('sent_on', 'desc')->first();
+        $previousBillingDate = $lastInvoiceDate ? $lastInvoiceDate->sent_on : date('Y-m-01');
+
+        return $previousBillingDate;
+    }
```
[the remark this answers is not in the document]
Comment (Gauravchamoli1997) [reply Modules/Client/Entities/Client.php:117]: Yes , we can directly take the last invoice and get the sent on date from there. That time I don't know about it so I create a Function to get previousBillingDate of invoice through database.

File: Modules/Invoice/Resources/views/index.blade.php [hunk 2]
```diff
@@ -254,24 +289,38 @@
...
+                                            config('client.service-rate-terms.per_resource.slug'))
+                                        {{ __('-') }}
                                     @else
                                         {{ $project->getBillableHoursForMonth($monthToSubtract) }}
+                                        @if (
+                                            $project->client->billingDetails->service_rate_term == config('client.service-rate-terms.per_hour.slug') ||
+                                                $project->client->billingDetails->service_rate_term == config('client.service-rate-terms.per_month.slug'))
+                                            <a class='text-danger' data-toggle="modal"
+                                                data-target="#modal-invoice-details{{ $project->id }}"
+                                                href="#"><big><i class="ml-10 fa fa-external-link-square "></i></big>
```
[the remark this answers is not in the document]
Comment (Gauravchamoli1997) [reply Modules/Invoice/Resources/views/index.blade.php:314]: @GauravGusain98 

> The alignment of the icon is not correct. Either it should be at right or a space after the amount image

I will Fix it.

**Comment by Gauravchamoli1997** [comment]
Is this look fine or I have to change this again
Screenshot (70)

**Pull request closed without merging**